hudi-common/src/main/java/org/apache/hudi/common/model/RecordPayloadType.java

```diff
@@ -94,19 +94,17 @@ public static RecordPayloadType fromClassName(String className) {
     return CUSTOM;
   }
 
-  public static String getPayloadClassName(HoodieConfig config) {
+  public static Option<String> getPayloadClassName(HoodieConfig config) {
     String payloadClassName;
     if (config.contains(PAYLOAD_CLASS_NAME)) {
       payloadClassName = config.getString(PAYLOAD_CLASS_NAME);
-    } else if (config.contains(PAYLOAD_TYPE)) {
-      payloadClassName = RecordPayloadType.valueOf(config.getString(PAYLOAD_TYPE)).getClassName();
     } else if (config.contains("hoodie.datasource.write.payload.class")) {
       payloadClassName = config.getString("hoodie.datasource.write.payload.class");
     } else {
-      payloadClassName = RecordPayloadType.valueOf(PAYLOAD_TYPE.defaultValue()).getClassName();
+      return Option.empty();
```

Comment:
Should this return the default payload class name?

Reply:
changed back